File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

float average(float a, float b);
float coleman(float a, float b);

int main(void)

{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    
    
    string text = get_string("Text:");
    
    
    for (int i = 0; i < strlen(text); i++)
    {
//Number of words
        if (text[i] == ' ')
        {
            words++;
        }
//Number of letters
        else if (text[i] >= 'a' && text[i] <= 'z')
        {
            letters++;
        }
        else if (text[i] >= 'A' && text[i] < 'Z')
        {
            letters++;
        }
//Number of symbols
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
//avarages and indexes
    float l = average(letters, words);
    float s = average(sentences, words);
    float index = coleman(l, s);
    int result = round(index);
//ifs
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", result);
    }
    else
    {
        printf("Grade 16+\n");
    }
    
}

float average(float a, float b)
{
    return (a * 100) / b;
}

float coleman(float a, float b)
{
    return (0.0588 * a - 0.296 * b - 15.8);
}

File: readability.py
from cs50 import get_string
# get text from user
text = get_string("string: ")
# convert text to lower case
text = text.casefold()
# initialize counters
letters = 0
sentences = 0
words = 0
# create alphabet list
Alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l',
            'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
# count letters
for i in range(0, 26):
    letters = letters + text.count(Alphabet[i])
# count words
words = text.count(" ") + 1
# create symbols list
Symbols = ['!', '.', '?']
# count sentences
for i in range(0, 3):
    sentences = sentences + text.count(Symbols[i])
# create l and s
l = (letters * 100) / words
s = (sentences * 100) / words
# finding Coleman-Liau formula result
grade = round(0.0588 * l - 0.296 * s - 15.8)
# if cases
if grade < 1:
    print("Before Grade 1")
elif 1 <= grade <= 16:
    print("Grade", grade)
else:
    print("Grade 16+")
